src/nvim/os/unix_defs.h

```diff
@@ -2,7 +2,7 @@
 #define NVIM_OS_UNIX_DEFS_H
 
 // Windows doesn't have unistd.h, so we include it here to avoid numerous
-// instances of `#ifdef HAVE_UNISTD_H'.
+// instances of `#ifdef WIN32'.
 #include <unistd.h>
```

Comment:
I guess in the strict sense yes.

But for platform includes which won't exist on all platforms or in the C standard (`<unistd.h>`/`<io.h>`) it seems to me they are better left in the platform specific headers?

Otherwise we have to do the same thing for Windows stuff too right? I mean:

`#ifdef HAVE_WINDOWS_H`, `#ifdef HAVE_IO_H` in the source files? That's just gross IMO.

I suppose we could do things like:

```c
#ifdef UNIX
# include <unistd.h>
#else
# include <io.h>
#endif
```

But since we are already going to have to include the OS def stuff anyway I don't see
that doing the above gains us any benefit. YMMV